security/pkg/stsservice/tokenmanagers/google/tokenmanager.go

```diff
@@ -0,0 +1,313 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package google
+
+import (
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+	"sync"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/duration"
+
+	"istio.io/istio/security/pkg/stsservice"
+	"istio.io/pkg/log"
+)
+
+const (
+	httpTimeOutInSec = 5
+	maxRequestRetry  = 5
+	contentType      = "application/json"
+	scope            = "https://www.googleapis.com/auth/cloud-platform"
+	tokenType        = "urn:ietf:params:oauth:token-type:access_token"
+	federatedToken   = "federated token"
+	accessToken      = "access token"
+)
+
+var (
+	tokenManagerLog        = log.RegisterScope("tokenManagerLog", "STS token manager debugging", 0)
+	federatedTokenEndpoint = "https://securetoken.googleapis.com/v1/identitybindingtoken"
+	// https://cloud.google.com/iam/docs/reference/credentials/rest/v1/projects.serviceAccounts/generateAccessToken
+	accessTokenEndpoint = "https://iamcredentials.googleapis.com/v1/projects/-/" +
+		"serviceAccounts/service-%s@gcp-sa-meshdataplane.iam.gserviceaccount.com:generateAccessToken"
```

Comment:
where dd this come from?

Reply:
This one refers to https://cloud.google.com/iam/docs/reference/credentials/rest/v1/projects.serviceAccounts/generateAccessToken, and I also had some offline discussions with cloud IAM team.

Reply:
So everyone will have a service account that is named exactly this? That doesn't seem correct and its not mentioned on the doc from what I can see

Reply:
I still don't understand why we would have a hardcoded service account here
This is so specific to your single use case yet it's exposed as some generic sts service?